baseapp-core/baseapp_core/channels.py

```diff
@@ -34,3 +36,72 @@ async def __call__(self, scope, receive, send):
             raise ValueError("User inactive or deleted")
 
         return await super().__call__(scope, receive, send)
+
+
+class JWTAuthMiddleware(BaseMiddleware):
+    def __init__(self, inner, *args, **kwargs):
+        from rest_framework_simplejwt.authentication import JWTAuthentication
+
+        super().__init__(inner, *args, **kwargs)
+        self._auth = JWTAuthentication()
+
+    @database_sync_to_async
+    def get_jwt_user_instance(self, token):
+        from rest_framework_simplejwt.exceptions import (
+            AuthenticationFailed,
+            InvalidToken,
+            TokenError,
+        )
+
+        if self._auth is None:
+            raise RuntimeError("JWTAuthMiddleware not initialized")
+
+        try:
+            validated_token = self._auth.get_validated_token(token)
+            return self._auth.get_user(validated_token)
+        except (InvalidToken, AuthenticationFailed, TokenError) as e:
+            logging.error(e)
+            return None
+
+    @database_sync_to_async
+    def refresh_access_token(self, refresh_token):
+        from rest_framework_simplejwt.exceptions import TokenError
+        from rest_framework_simplejwt.tokens import RefreshToken
+
+        try:
+            refresh = RefreshToken(refresh_token)
+            return str(refresh.access_token)
+        except TokenError as e:
+            logging.error(e)
+            return None
+
+    async def __call__(self, scope, receive, send):
+        if "Authorization" not in scope["subprotocols"]:
```

Comment:
@anicioalexandre so this will be called for any authed async requests (like websockets)?

Reply:
@Janekk correct, for any connection that uses the `JWTAuthMiddleware`, like [here](https://bitbucket.org/silverlogic/baseapp-django-v3/pull-requests/45/diff#chg_apps/asgi.py_newline27)